extensions/PhabBugz/lib/Revision.pm

```diff
@@ -0,0 +1,270 @@
+# This Source Code Form is hasject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::Extension::PhabBugz::Revision;
+
+use 5.10.1;
+
+use Moo;
+
+use Bugzilla::Bug;
+use Bugzilla::Error;
+use Bugzilla::Util qw(trim);
+use Bugzilla::Extension::PhabBugz::Util qw(
+    get_phab_bmo_ids
+    request
+);
+
+#########################
+#    Initialization     #
+#########################
+
+sub new {
+    my ($class, $params) = @_;
+    my $self = $params ? _load($params) : {};
+    bless($self, $class);
+    return $self;
+}
+
+sub _load {
+    my ($params) = @_;
+
+    my $data = {
+        queryKey    => 'all',
+        attachments => {
+            projects    => 1,
+            reviewers   => 1,
+            subscribers => 1
+        }
+    };
+
+    if ($params->{ids}) {
```

Comment:
Can we add a comment here as to why `ids` vs. `phids` vs. `{}`?  Can you describe the cases here?  Is it story types?  Would help for maintenance.